### PR TITLE
Zone Record Regions

### PR DESCRIPTION
**Pull request by jodosha**
@weppos @aeden Please review. Thanks!

### CODE REVIEW

File: dnsimple/zones_records_test.go [hunk 1]
```diff
@@ -90,28 +93,37 @@ func TestZonesService_CreateRecord(t *testing.T) {
 		testMethod(t, r, "POST")
 		testHeaders(t, r)
 
-		want := map[string]interface{}{"name": "foo", "content": "192.168.0.10", "type": "A"}
-		testRequestJSON(t, r, want)
+		// --- FAIL: TestZonesService_CreateRecord (0.00s)
```
Comment (jodosha) [inline dnsimple/zones_records_test.go]: @weppos I [declared](https://github.com/dnsimple/dnsimple-go/pull/47/files#diff-3df835be8edaab5a6734faa367081eeaR13) `Region.Name` as `omitempty`, but it still forced as key in the request payload for create/update.

I commented out this assertion because there is something magical ✨  happening here. Can you please check? Thanks.

File: dnsimple/zones_records.go [hunk 2]
```diff
@@ -6,17 +6,18 @@ import (
...
-	TTL          int    `json:"ttl,omitempty"`
-	Priority     int    `json:"priority,omitempty"`
-	SystemRecord bool   `json:"system_record,omitempty"`
-	CreatedAt    string `json:"created_at,omitempty"`
-	UpdatedAt    string `json:"updated_at,omitempty"`
+	ID           int      `json:"id,omitempty"`
+	ZoneID       string   `json:"zone_id,omitempty"`
+	ParentID     int      `json:"parent_id,omitempty"`
+	Type         string   `json:"type,omitempty"`
+	Name         string   `json:"name",omitempty`
```
Comment (weppos) [inline dnsimple/zones_records.go]: @jodosha why did you change name to add `omitempty`?

File: dnsimple/zones_records_test.go (hunk 1 above)
Comment (weppos) [inline dnsimple/zones_records_test.go]: @jodosha it's a type issue. I was able to reproduce it

```
➜  dnsimple-go git:(zone-record-regions) go test ./...
--- FAIL: TestZonesService_CreateRecord (0.00s)
	dnsimple_test.go:68: Request parameters = map[type:MX name: content:mxa.example.com regions:[SV1 IAD]], want map[content:mxa.example.com type:MX regions:[SV1 IAD]]
FAIL
FAIL	github.com/dnsimple/dnsimple-go/dnsimple	0.072s
ok  	github.com/dnsimple/dnsimple-go/dnsimple/webhook	0.020s
```

changing the [format of the response](https://golang.org/pkg/fmt/) from `%v` to `%#v` revealed the issue

```
--- FAIL: TestZonesService_CreateRecord (0.00s)
	dnsimple_test.go:68: Request parameters = map[string]interface {}{"type":"MX", "name":"", "content":"mxa.example.com", "regions":[]interface {}{"SV1", "IAD"}}, want map[string]interface {}{"content":"mxa.example.com", "type":"MX", "regions":[]string{"SV1", "IAD"}}
FAIL
FAIL	github.com/dnsimple/dnsimple-go/dnsimple	0.064s
ok  	github.com/dnsimple/dnsimple-go/dnsimple/webhook	0.015s
```

File: dnsimple/zones_records.go (hunk 2 above)
Comment (weppos) [inline dnsimple/zones_records.go]: Also note the sintax is incorrect, it should be

    `json:"name,omitempty"`

not

    `json:"name",omitempty`

However, that should not be changed.

Comment (jodosha) [inline dnsimple/zones_records.go]: Because it is mandatory for record creation, but not for its update.

**Comment by weppos** [comment]
@jodosha I've fixed the issue. I'll deal with #33 in a different PR.

**Comment by jodosha** [comment]
@weppos cool, thank you.
